### PR TITLE
Cached LoggingWidget improvement and Removing Logging TextBoxes

### PR DESCRIPTION
**Pull request by inforithmics**
Cached LoggingWidget improvement by updating the map of the cached logging widget
And removed Logging Textboxes from samples

see https://github.com/Mapsui/Mapsui/pull/2377#issuecomment-1879821981

### CODE REVIEW

**Comment by charlenni** [comment]
I will look into this in the next days. But not before the Widget rework isn't done. The greater idea is to remove Map from all widgets. I'm not sure, if this is possible.

Would it be possible to to remove the widget from rendering?

**Comment by charlenni** [comment]
Would it be easier to create the LoggingWidget in the constructor for the Map instead of MapControl?

**Comment by inforithmics** [comment]
The setting of the map isn't necessary with the widget improvements pull request. But the widget creation should stay in the current position because only when a debugger is attached the widgetlogging is created.

File: Mapsui.UI.Shared/MapControl.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public partial class MapControl : INotifyPropertyChanged, IDisposable
     // keeps track of the widgets count to see if i need to recalculate the touchable widgets.
     private int _updateTouchableWidget;
     // Default LoggingWidget when the debugger is attached
-    private LoggingWidget _loggingWidget;
+    private LoggingWidget? _loggingWidget;
```
Comment (pauldendulk) [inline Mapsui.UI.Shared/MapControl.cs:71]: I already saw the warning in the build log. Good fix.

**Comment by pauldendulk** [comment]
> Would it be easier to create the LoggingWidget in the constructor for the Map instead of MapControl?

I would prefer that. I expect we could still check if the debugger is attached if we do it like that.

File: Mapsui.UI.Shared/MapControl.cs [hunk 2]
```diff
@@ -344,6 +344,8 @@ private void CheckForLoggingWidget(Map map)
                     Opacity = 0.0f,
                     LogLevelFilter = LogLevel.Trace,
                 };
+            else
+                _loggingWidget.Map = map;
```
Comment (pauldendulk) [inline Mapsui.UI.Shared/MapControl.cs:348]: As mentioned elsewhere we should come up with a different mechanism for refresh, but this is the fix in the current state.

File: Mapsui/Widgets/LoggingWidget/LoggingWidget.cs [hunk 3]
```diff
@@ -283,6 +283,15 @@ public Color InformationTextColor
         }
     }
 
+    public Map Map
```
Comment (pauldendulk) [inline Mapsui/Widgets/LoggingWidget/LoggingWidget.cs:286]: Ugly but necessary for now.

**Review by pauldendulk — APPROVED** [review]
All good improvements. I want to review smaller merge requests. This is small merge request, great. It would be even better if it was still smaller. Just removing the old logging widget would have resulted in no discussion at all, such a PR could be merged from the phone in the train.

**Comment by pauldendulk** [comment]
> I will look into this in the next days. But not before the Widget rework isn't done.

I am going to merge the PR. This is small PR, that is a good thing. By blocking a smaller PR with a bigger PR we go in the opposite direction.

**Pull request merged**